src/pypowsybl_jupyter/selectcontext.py

```diff
@@ -54,3 +57,15 @@ def get_filtered_vls_as_list(self):
     def extend_filtered_vls(self, id):
         if (id in self.vls.index) and (id not in self.vls_filtered.index):
             self.vls_filtered = pd.concat([self.vls_filtered, self.vls.loc[[id]]])
+
+    def add_to_history(self, id):
+        if (id in self.vls.index):
+            row_to_add = self.vls.loc[id].to_dict()
+            for item in self.history:
+                if item['id'] == row_to_add['id']:
+                    self.history.remove(item)
+                    break
```

Comment:
It's ok to have an item several times in the history (like in a browser history), so I'd remove that